distribution/pull_v2_unix.go

```diff
@@ -5,9 +5,14 @@ package distribution
 import (
 	"github.com/docker/distribution"
 	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/manifest/schema2"
+	"github.com/docker/docker/distribution/xfer"
 )
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
+	if ld.src.MediaType == schema2.MediaTypeForeignLayer {
```

Comment:
There's been multiple reports on https://hub.docker.com/r/microsoft/nanoserver/ and https://hub.docker.com/r/microsoft/windowsservercore/ as well as other Windows based images of users attempting to pull Windows images on Linux daemons, and the error message returned (unknown blob) is not at all helpful to understand the problem.

This is a temporary solution until support for foreign layers is implemented, but it would help users encountering this error understand why the pull is failing (and it also stops the useless retries, which make it seem like a temporary error rather than a platform compatibility issue).